### PR TITLE
Parallel Chapel pidigits implementation

### PR DESCRIPTION
**Pull request by rivalq**
I have used [this](https://benchmarksgame-team.pages.debian.net/benchmarksgame/program/pidigits-gpp-6.html)  C++ program to create a Parallel version in Chapel.

### CODE REVIEW

**Comment by rivalq** [comment]
@bradcray, I have drafted the PR.

File: util/quickstart/setchplenv.bash [hunk 1]
```diff
@@ -58,8 +58,8 @@ export CHPL_TASKS=fifo
 echo "Setting CHPL_MEM to cstdlib"
 export CHPL_MEM=cstdlib
 
-echo "Setting CHPL_GMP to none"
-export CHPL_GMP=none
+echo "Setting CHPL_GMP to system"
+export CHPL_GMP=system
```
Comment (bradcray) [inline util/quickstart/setchplenv.bash:62]: I don't think this change was meant to be in this PR.

**Comment by bradcray** [comment]
Hi @rivalq — It looks to me as though these two files were added to the PR accidentally.  Does that seem right?  If not, what do they do?

* test/studies/shootout/pidigits/rivalq/.pidigits-rivalq.good.kate-swp
* test/studies/shootout/pidigits/rivalq/pidigits2.chpl

File: util/quickstart/setchplenv.bash (hunk 1 above)
Comment (rivalq) [reply util/quickstart/setchplenv.bash:62]: Ohh, Actually it was for my local system to use `BigInt`  library.

**Comment by rivalq** [comment]
> Hi @rivalq — It looks to me as though these two files were added to the PR accidentally. Does that seem right? If not, what do they do?
> 
> * test/studies/shootout/pidigits/rivalq/.pidigits-rivalq.good.kate-swp
> * test/studies/shootout/pidigits/rivalq/pidigits2.chpl

I don't know about the first file, it must have been generated while I was using `kate` text editor.
And the second file may have been accidentally copied with other `pidigits2*` files.

**Comment by rivalq** [comment]
Hi @bradcray 

I actually have commented on both the issues on that same day, I don't know why they are not visible to you. 

1 . Change in `setchplenv.bash` is just for my local system to use the `BigInt` library. We can revert that change.
2.  That `.kate` file is I believe temporary file created by my system's default text editor. and about the `pidigits2.chpl` file, you are correct, it was accidentally copied with other `pidigits2*`  files.

**Comment by bradcray** [comment]
@rivalq: This looks good to merge to me.  I won't have time to deal with any fallout in testing tomorrow, so will plan to merge it tomorrow rather than today in order to triage any problems on Friday.

**Comment by rivalq** [comment]
@bradcray 

Is there any Chapel vs C++ performance analysis tool?

**Comment by bradcray** [comment]
Hi @rivalq — No, there's no high-level tool that supports C++ vs. Chapel comparisons, so the two main techniques would be to either:

* put timers into the code around key sections of the computation to see where time is spent in the two versions
* or potentially applying a normal C profiling tool to the C code generated by Chapel when using its C back-end

One thing that makes the second approach challenging at times is that it's not generally possible for such tools to distinguish between useful Chapel work and threads spinning waiting for a task to run.

I'd hoped to spend more time on this last month myself, but have not found the time to get back to it yet.  Hopefully this is obvious, but I'm not counting on you to do more with it unless you're interested.  I would like to see Chapel return to the fastest entry on the CLBG site for pidigits, but I don't know how much work it will be to understand what the current gap is between Chapel and C/C++.

**Comment by bradcray** [comment]
Hi @rivalq — I've gotten fired up about the CLBG site again lately ([we had some nice improvements last week](https://twitter.com/ChapelLanguage/status/1486072565974003712)), and so finally got back to looking at your port of the pidigits benchmark today.  I ended up writing a new version that is a hybrid between the one you submitted and the gcc#6 version (which is also based on the gpp#6 version that yours was) and came up with something that I think is worth submitting to the site, perhaps next week.  I would like to give you credit on it as well since I learned from what you'd done in this PR if that sounds good to you.  And if so, should I use the name Jatin Garg from your GitHub profile?  Thanks very much... it would've taken me much longer to do this port without your version to learn from.

**Pull request merged**